aspnetcore/mvc/views/razor.md

```diff
@@ -15,74 +15,74 @@ uid: mvc/views/razor
 
 By [Taylor Mullen](https://twitter.com/ntaylormullen) and [Rick Anderson](https://twitter.com/RickAndMSFT)
 
-## What is Razor?
-
-Razor is a markup syntax for embedding server based code into web pages. The Razor syntax consists of Razor markup, C# and HTML. Files containing Razor generally have a *.cshtml* file extension.
+Razor is a markup syntax for embedding server-based code into webpages. The Razor syntax consists of Razor markup, C#, and HTML. Files containing Razor generally have a *.cshtml* file extension.
 
 ## Rendering HTML
 
-The default Razor language is HTML. Rendering HTML from Razor is no different than in an HTML file. A Razor file with the following markup:
+The default Razor language is HTML. Rendering HTML from Razor markup is no different than rendering HTML from an HTML file. If you place the following markup into a Razor file, it's rendered by the server unchanged:
 
-```html
+```cshtml
 <p>Hello World</p>
 ```
 
-Is rendered unchanged as `<p>Hello World</p>` by the server.
+```html
+<p>Hello World</p>
+```
 
 ## Razor syntax
 
-Razor supports C# and uses the `@` symbol to transition from HTML to C#. Razor evaluates C# expressions and renders them in the HTML output. Razor can transition from HTML into C# or into Razor-specific markup. When an `@` symbol is followed by a [Razor reserved keyword](#razor-reserved-keywords) it transitions into Razor-specific markup, otherwise it transitions into plain C#.
+Razor supports C# and uses the `@` symbol to transition from HTML to C#. Razor evaluates C# expressions and renders them in the HTML output.
 
-<a name=escape-at-label></a>
+When an `@` symbol is followed by a [Razor reserved keyword](#razor-reserved-keywords), it transitions into Razor-specific markup. Otherwise, it transitions into plain C#.
 
-HTML containing `@` symbols may need to be escaped with a second `@` symbol. For example:
+`@` symbols are escaped by an `@` symbol:
 
 ```cshtml
 <p>@@Username</p>
 ```
 
-would render the following HTML:
+The code is rendered in HTML with a single `@` symbol:
 
-```cshtml
+```html
 <p>@Username</p>
 ```
 
-<a name=razor-email-ref></a>
-
-HTML attributes and content containing email addresses don’t treat the `@` symbol as a transition character.
+HTML attributes and content containing email addresses don't treat the `@` symbol as a transition character. The email addresses in the following example are untouched by Razor parsing:
 
-   `<a href="mailto:Support@contoso.com">Support@contoso.com</a>`
+```cshtml
+<a href="mailto:Support@contoso.com">Support@contoso.com</a>
+```
 
 ## Implicit Razor expressions
 
-Implicit Razor expressions start with `@` followed by C# code. For example:
+Implicit Razor expressions start with `@` followed by C# code:
 
-```html
+```cshtml
```

Comment:
Does the markup understand cshtml?

Reply:
@NTaylorMullen Yes, cshtml is supported (this was a recent change).